### PR TITLE
Test module loader

### PR DESCRIPTION
**Pull request by izrik**
This PR adds tests for the `ModuleLoader` class, in the `blueflood-core` module. There are already tests for it in the `blueflood-http` module, but since that's different than the module that the class is defined in, coverage is being reported as 0% (see [here](https://coveralls.io/builds/5442321/source?filename=blueflood-core%2Fsrc%2Fmain%2Fjava%2Fcom%2Frackspacecloud%2Fblueflood%2Futils%2FModuleLoader.java)). Also, these tests better demonstrate the behavior of the class, and provide pertinent negative test cases.

According to JaCoCo, the test class, by itself, provides coverage for `ModuleLoader` to the tune of:
* 75% for instructions 
* 100% for branches
* 78% for lines
What's not tested is exception handling and the default parameter-less constructor.

On the whole, `ModuleLoader` appears to be a thin, confusing, and broken wrapper around `ClassLoader.loadClass(String)`. I believe it should be refactored away.

### CODE REVIEW

**Comment by izrik** [comment]
@VinnyQ @ChandraAddala @shintasmith @usnavi Please review and comment. I'll merge once I get +1 and any concerns that come up are addressed.

File: blueflood-core/src/test/java/com/rackspacecloud/blueflood/utils/ModuleLoaderTest.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                CoreConfig.DISCOVERY_MODULES,
+                "com.rackspacecloud.blueflood.io.DiscoveryIO");
+
+        // when
+        Object loadedModule = ModuleLoader.getInstance(DiscoveryIO.class, CoreConfig.DISCOVERY_MODULES);
+
+        // then
+        Assert.assertNull(loadedModule);
+    }
+}
```
Comment (ChandraAddala) [inline blueflood-core/src/test/java/com/rackspacecloud/blueflood/utils/ModuleLoaderTest.java]: Can you also test a scenario where the class does not have a default(no-arg) constructor?

**Comment by ChandraAddala** [comment]
Looks good.

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/utils/ModuleLoader.java [hunk 2]
```diff
@@ -36,7 +37,7 @@ public static Object getInstance(Class c, CoreConfig moduleName) {
         List<String> modules = Configuration.getInstance().getListProperty(moduleName);
         if (modules.isEmpty())
             return null;
-        if (!modules.isEmpty() && modules.size() != 1) {
+        if (modules.size() != 1) {
```
Comment (ChandraAddala) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/utils/ModuleLoader.java]: Good refactoring.

File: blueflood-core/src/test/java/com/rackspacecloud/blueflood/utils/ModuleLoaderTest.java (hunk 1 above)
Comment (ChandraAddala) [inline blueflood-core/src/test/java/com/rackspacecloud/blueflood/utils/ModuleLoaderTest.java]: Also for a class which is package-protected and is from a different package than the test case.

Comment (izrik) [inline blueflood-core/src/test/java/com/rackspacecloud/blueflood/utils/ModuleLoaderTest.java]: @ChandraAddala Done and done.

**Comment by izrik** [comment]
@ChandraAddala How does it look now?

File: blueflood-core/src/test/java/com/rackspacecloud/blueflood/utils/ModuleLoaderTest.java [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+        // when
+        Object loadedModule = ModuleLoader.getInstance(DiscoveryIO.class, CoreConfig.DISCOVERY_MODULES);
+
+        // then
+        Assert.assertNotNull(loadedModule);
+        Assert.assertEquals(DummyDiscoveryIO3.class, loadedModule.getClass());
+    }
+
+    @Test
+    public void singleClassInDifferentPackageYieldsThatClass() {
```
Comment (ChandraAddala) [inline blueflood-core/src/test/java/com/rackspacecloud/blueflood/utils/ModuleLoaderTest.java]: I am thinking of a test case where it would throw "IllegalAccessException". I believe for that we have to pick a class from a different package which is at package-protected access level or may be a private class to trigger that exception.

**Comment by izrik** [comment]
@ChandraAddala How about now?

**Comment by ChandraAddala** [comment]
Looks good.
